visma/io/parser.py

```diff
@@ -221,6 +222,18 @@ def tokensToString(tokens):
                 if token.power != 1:
                     tokenString += "^" + "(" + str(token.power) + ")"
                 tokenString += "(" + tokensToString([token.operand]) + ")"
+        elif isinstance(token, Sine) or isinstance(token, Cosine) or isinstance(token, Tangent) or isinstance(token, Cosecant) or isinstance(token, Secant) or isinstance(token, Cotangent):
```

Comment:
Create a superclass trigonometric and make sin, cos, etc as sub-classes. 
```suggestion
        elif isinstance(token, Trigonometric):
```